cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/worker/TaskWorkerServiceLauncher.java

```diff
@@ -195,6 +194,24 @@ public void run() {
     this.twillController = activeController;
   }
 
+  private SecurityContext createSecurityContext() {
+    SecurityContext.SecurityContextBuilder builder = new SecurityContext.SecurityContextBuilder();
+    String twillUserIdentity = cConf.get(Constants.Twill.Security.IDENTITY_USER);
+    if (twillUserIdentity != null) {
+      builder.withIdentity(twillUserIdentity);
+    }
+
+    try {
+      builder.withUserId(cConf.getLong(Constants.TaskWorker.CONTAINER_RUN_AS_USER));
+    } catch (NullPointerException e) {
```

Comment:
what returns NPE? Also in that case should we ignore the exception?

Reply:
cConf.getLong returns NPE if configuration does not exist.

Reply:
If the default cannot be defined in `cdap-default.xml`, then you should use the `public long getLong(String name, long defaultValue)` method. Do you have a reasonable default value that can be set in the cdap-default.xml